Library/Homebrew/exceptions.rb

```diff
@@ -435,11 +435,23 @@ def initialize(flags)
 # the compilers available on the user's system
 class CompilerSelectionError < RuntimeError
   def initialize(formula)
-    super <<-EOS.undent
-      #{formula.full_name} cannot be built with any available compilers.
-      To install this formula, you may need to:
-        brew install gcc
-      EOS
+    if MacOS.version > :tiger
+      super <<-EOS.undent
+        #{formula.full_name} cannot be built with any available compilers.
+        To install this formula, you may need to:
+          brew install gcc
+        EOS
```

Comment:
The main thing there is the order: on 10.4 it's more useful to suggest apple-gcc42 first (since it can build ObjC and a few other Apple-specific features).